python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -252,19 +247,30 @@ async def update_status(self):
                     "startAfterPowerLoss",
                     "startAfterDryerPreRun",
                 ],
-                status.registers[2],
+                status[2],
             ),
         )
 
+        self._start_by_remote = status[2] & 0x01 == 0x01
+        self._status_update = True
+
+        if status[0] & 0x02 == 0x02:
+            # None can be passed, as begin_enable and begin_disable called from
+            # _do_change_state don't care about its content
+            if self.summary_state != salobj.State.ENABLED:
+                await self.do_enable(None)
+                self.log.info("Auto switched to enabled, as compressor is running")
+        else:
+            if self.summary_state != salobj.State.DISABLED:
```

Comment:
Same comment here. You cannot change the CSC state based on whether the compressor is running, with one exception: you can always go to FAULT if something isn't right.

I assume this handles the case that the compressor stops for some reason (e.g. manual override or it gives up for some reason)? If so, going to FAULT is probably the best solution. I can't really think what else to do. You surely don't want to stay in ENABLED state (assuming ENABLED = compressor running).

Reply:
DM-35280